spec/cancan/model_adapters/active_record_adapter_spec.rb

```diff
@@ -115,6 +115,17 @@ class User < ActiveRecord::Base
       expect(Article.accessible_by(@ability)).to eq([article])
     end
 
+    it 'does not fires query with accessible_by() for abilities defined with association' do
+      user = User.create!
+      @ability.can(:read, Article, user.articles) { true }
```

Comment:
this is not how you write rules in cancancan. you should replace it with `can :read, Article, user_id: user.id`, as discussed also #552.